config/config.go

```diff
@@ -187,27 +187,31 @@ func (c *Config) Validate() error {
 		destinationChains.Add(s.ChainID)
 	}
 
-	return nil
-}
-
-// GetSourceIDs returns the Subnet and Chain IDs of all subnets configured as a source
-func (cfg *Config) GetSourceIDs() ([]ids.ID, []ids.ID, error) {
-	var sourceSubnetIDs []ids.ID
-	var sourceChainIDs []ids.ID
-	for _, s := range cfg.SourceSubnets {
-		subnetID, err := ids.FromString(s.SubnetID)
-		if err != nil {
-			return nil, nil, fmt.Errorf("invalid subnetID in configuration. error: %v", err)
+	// Validate the source chains, and validate that the allowed destinations are configured as destinations
+	sourceChains := set.NewSet[string](len(c.SourceSubnets))
+	for _, s := range c.SourceSubnets {
+		if err := s.Validate(); err != nil {
+			return err
 		}
-		sourceSubnetIDs = append(sourceSubnetIDs, subnetID)
+		if sourceChains.Contains(s.ChainID) {
+			return fmt.Errorf("configured source subnets must have unique chain IDs")
+		}
+		sourceChains.Add(s.ChainID)
 
-		chainID, err := ids.FromString(s.ChainID)
-		if err != nil {
-			return nil, nil, fmt.Errorf("invalid subnetID in configuration. error: %v", err)
+		for _, blockchainID := range s.SupportedDestinations {
```

Comment:
thoughts on moving this to `sourceChain.Validate()`? In my mind these checks are still validating the source chain configs, we can pass in the destination chains to validate.

Reply:
Good call, moved.